### PR TITLE
#1426 Notifications Lambda

### PR DESCRIPTION
**Pull request by mannykary**
### What does this pull request aim to achieve? 💯

Notifications Lambda

_Please be concise and link any related issue(s):_

- [x] I have checked that no one else is working on similar changes.
- [x] I have read the latest [**README**](README.md) and understand the development workflow.
- [x] The name of this branch is: **`feature/<branch_name>`**, or **`hotfix/<branch_name>`** if this is a hotfix, where `<branch_name>` briefly describes the issue(s) resolved and is prefixed with the issue number(s) (e.g. `1127-1130-update-git-branching-model`). The branch is created in a cloned version of this repo, **not a forked repo**.
- [x] This branch is rebased/merged with the **latest staging** (or the **latest production** if it is a hotfix).
- [x] There are no merge conflicts.
- [x] There are no console warnings and errors.
- [x] On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".
- [x] The title describes the issue(s) being addressed using format: **`<issue_numbers> - <brief_issue_description>`** (e.g. `#1127/#1130 - Update Git Branching Model`)

### CODE REVIEW

File: services/notification/src/helpers/template-builder.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+          text: this._loadTemplateFile("../templates/instant/comment.txt"),
+        },
+        like: {
+          html: this._loadTemplateFile("../templates/instant/like.html"),
+          subject: "{{triggeredBy.name}} liked your post: {{post.title}}",
+          text: this._loadTemplateFile("../templates/instant/like.txt"),
+        },
+        share: {
+          html: this._loadTemplateFile("../templates/instant/share.html"),
+          subject: "{{triggeredBy.name}} shared your post: {{post.title}}",
```
Comment (Zaydme) [inline services/notification/src/helpers/template-builder.js:30]: @mannykary a new field was added to the model, `sharedVia` which is a string, the name of the medium used to share the post.

File: services/notification/deploy/lambda.tf [hunk 2]
```diff
@@ -1,3 +1,12 @@
+  locals {
+    base_url = {
+      review      = "https://feature1427-notifications-unsubscribe.fightpandemics.xyz" # Point to Dawei's branch for now
```
Comment (ddveloper) [inline services/notification/deploy/lambda.tf]: that one is closed, may need an update here?

File: services/notification/src/config.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    port: Number(process.env.EXTERNAL_LOGGER_PORT),
+  },
+  mailService: {
+    apiKeyId: process.env.SES_AWS_ACCESS_KEY_ID,
+    apiKey: process.env.SES_AWS_SECRET_ACCESS_KEY,
+    fromEmailAddress: process.env.FROM_EMAIL_ADDRESS,
+    region: process.env.SES_AWS_REGION,
+  },
+  nodeEnv: process.env.NODE_ENV,
+  unsubscribeTokenKey: process.env.UNSUBSCRIBE_TOKEN_KEY,
```
Comment (ddveloper) [inline services/notification/src/config.js:34]: the key is changed back to use `AUTH_SECRET_KEY`, may need an update here

**Review by Zaydme — APPROVED** [review]
Looks good 👍

File: services/notification/deploy/lambda.tf (hunk 2 above)
Comment (mannykary) [reply services/notification/deploy/lambda.tf]: Yes, I'll update it to point to the feature notifications build

File: services/notification/src/helpers/template-builder.js (hunk 1 above)
Comment (mannykary) [reply services/notification/src/helpers/template-builder.js:30]: Got it, will add it.

File: services/notification/src/config.js (hunk 3 above)
Comment (mannykary) [reply services/notification/src/config.js:34]: I think this is right; the environment variable name is different here but it is using the same key as in the backend; see the environment variable set on the Lambda [here](https://github.com/FightPandemics/FightPandemics/blob/d1b925934cf3d0149a23ad61c460ffecaa955182/services/notification/deploy/lambda.tf#L45), and the AWS SSM parameter being pulled [here](https://github.com/FightPandemics/FightPandemics/blob/d1b925934cf3d0149a23ad61c460ffecaa955182/services/notification/deploy/variables.tf#L41), which is the same as what you used (the Auth0 client secret key).

Backend config variable your code pulls from is `auth.secretKey` [here](https://github.com/FightPandemics/FightPandemics/blob/staging/backend/config.js#L56)
Which comes from `AUTH_SECRET_KEY` environment variable which is injected into the backend [here](https://github.com/FightPandemics/FightPandemics/blob/staging/main.tf#L131)
Which comes from the AWS SSM parameter pulled [here](https://github.com/FightPandemics/FightPandemics/blob/staging/main.tf#L39)
Which is the same AWS SSM parameter as what I'm using to generate the unsubscribe token.

**Pull request merged**